internal/supervisor/import.go

```diff
@@ -134,6 +134,22 @@ func (s *ImportSupervisor) Do() error {
 	err = s.importTranslation(work)
 	if err != nil {
 		s.logger.WithError(err).Errorf("failed to perform work on Import %s", work.ID)
+		workError := err.Error()
+		go func() {
```

Comment:
So this looks a little scary to me as a potential place for a memory leak. This probably does not apply right now as we do not expect many imports and even fewer of them (if any) to hit this edge case. I am just generally worried of unmanaged things running infinitely. That said I get your point about not investing time in some complex mechanism for an edge case we might never encounter. 

We could possibly consider some fixed number of retries with exponential backoff if we want to handle cases when service will be down for a significant amount of time but as mentioned above it might not apply at all for the case of imports so we probably will be fine with that.

Anyway, just giving some voice to my software paranoia 😛  Leaving it up to you 😉

Reply:
Added exponential backoff in this [PR](https://github.com/mattermost/mattermost-cloud/pull/811)